src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyCommandRunner.cs

```diff
@@ -21,62 +22,80 @@ public class VerifyCommandRunner : IVerifyCommandRunner
     {
         public async Task<int> ExecuteCommandAsync(VerifyArgs verifyArgs)
         {
-            if (verifyArgs.Type != VerifyArgs.VerificationType.Signatures)
+            var executedSuccessfully = false;
+            
+            if (ShouldExecuteType(verifyArgs, VerificationType.Signatures))
             {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.VerifyCommand_VerificationTypeNotSupported));
-            }
-
-            var packagesToVerify = LocalFolderUtility.ResolvePackageFromPath(verifyArgs.PackagePath);
-            LocalFolderUtility.EnsurePackageFileExists(verifyArgs.PackagePath, packagesToVerify);
+                var packagesToVerify = LocalFolderUtility.ResolvePackageFromPath(verifyArgs.PackagePath);
+                LocalFolderUtility.EnsurePackageFileExists(verifyArgs.PackagePath, packagesToVerify);
 
-            var trustProviders = SignatureVerificationProviderFactory.GetSignatureVerificationProviders();
-            var verifier = new PackageSignatureVerifier(trustProviders, SignedPackageVerifierSettings.RequireSigned);
+                var trustProviders = SignatureVerificationProviderFactory.GetSignatureVerificationProviders();
+                var verifier = new PackageSignatureVerifier(trustProviders, SignedPackageVerifierSettings.RequireSigned);
 
-            var errorCount = 0;
+                var errorCount = 0;
 
-            foreach (var package in packagesToVerify)
-            {
-                try
+                foreach (var package in packagesToVerify)
                 {
-                    errorCount += await VerifyPackageAsync(package, verifyArgs.Logger, verifier);
-                }
-                catch (InvalidDataException)
-                {
-                    verifyArgs.Logger.LogError(string.Format(CultureInfo.CurrentCulture, Strings.VerifyCommand_PackageIsNotValid, package));
+                    try
+                    {
+                        errorCount += await VerifySignatureForPackageAsync(package, verifyArgs.Logger, verifier);
+                    }
+                    catch (InvalidDataException e)
+                    {
+                        verifyArgs.Logger.LogError(string.Format(CultureInfo.CurrentCulture, Strings.VerifyCommand_PackageIsNotValid, package));
+                        verifyArgs.Logger.LogDebug(string.Format(CultureInfo.CurrentCulture, Strings.Error_CommandFailedWithException, nameof(VerifySignatureForPackageAsync), e.Message));
```

Comment:
Don't localize this just log the full stack if it would be useful, also use `ExceptionUtilities` if it makes sense here